Source/WebCore/rendering/svg/SVGResourcesCache.cpp

```diff
@@ -108,7 +108,7 @@ static inline bool rendererCanHaveResources(RenderObject& renderer)
 
 void SVGResourcesCache::clientStyleChanged(RenderElement& renderer, StyleDifference diff, const RenderStyle* oldStyle, const RenderStyle& newStyle)
 {
-    if (diff == StyleDifference::Equal || !renderer.parent())
```

Comment:
Do we need to special case the filter resource only?
```
if ((!renderer.isSVGResourceFilterPrimitive() && diff == StyleDifference::Equal) || !renderer.parent())
```
Do we need to add a comment here to say why it is okay to proceed even if diff is StyleDifference::Equal?

Reply:
Yes, I think that's a good idea to continue returning early for `StyleDifference::Equal` when this is a primitive.